rules/precompiled_apple_resource_bundle.bzl

```
@@ -161,6 +161,17 @@ def _precompiled_apple_resource_bundle_impl(ctx):
 
     # Create a file for bundletool to know what files to copy
     # https://github.com/bazelbuild/rules_apple/blob/d29df97b9652e0442ebf21f1bc0e04921b584f76/tools/bundletool/bundletool_experimental.py#L29-L46
+    #
+    # ps: removing `.momd` files from here since they'll be collected in `AppleResourceInfo` => `datamodels`
+    # below and added under the `.bundle` directory like other resources. This is so data model files
+    # are visible to other rules looking for information in the `AppleResourceInfo` provider. See this
+    # PR for more context: https://github.com/bazel-ios/rules_ios/pull/750
+    control_files = [
+        f
+        for f in control_files
+        if hasattr(f, "src") and not f.src.endswith(".momd")
+    ]
+
```

NIT: Feels like the comments are out of order now and could be a bit more terse. Also, bikeshedding a bit, I think the PR information can generally be left out, as it can be traced back from git commit metadata when useful, but usually doesn't add much value and reduced readability.

```suggestion
    # `*.momd` files will be collected in `AppleResourceInfo` below and added under the `.bundle` directory like other resources.
    control_files = [
        f
        for f in control_files
        if hasattr(f, "src") and not f.src.endswith(".momd")
    ]

    # Create a file for bundletool to know what files to copy
    # https://github.com/bazelbuild/rules_apple/blob/d29df97b9652e0442ebf21f1bc0e04921b584f76/tools/bundletool/bundletool_experimental.py#L29-L46
```

---

Makes sense to me, def being more verbose here than necessary. These changes are still not right though and I'll continue to investigate. The failures in the latest run show me that something is missing since a test that was passing before is not anymore.

Will make the changes once I root cause that.